### PR TITLE
Fix lookup and validation

### LINKED ISSUE #1170: Settings: first character of username text input cannot be changed
**Describe the bug**
User is unable to backspace the first character of username text input after initial entry

**To Reproduce**
1. Log in to Files with an account that has no username set
2. Navigate to settings
3. Navigate to the profile tab
4. Click "Add a username" 
5. Enter a long unique string as input
6. Attempt to remove the input that you typed

**Observed behavior**
It's not possible to remove the first character that you typed, you have to navigate away and back to the tab in order to set the username to something else. 

**Expected behavior**
It should be possible to remove all prior input text 

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser Chrome
 - Version 91

**Additional context**
Dev branch only, set username feature is not available on production.

### PR DESCRIPTION
**Pull request by Tbaut**
closes #1170 and also most of #1159 since the change was merged on the api, just waiting for the api-spec update.
edit: api-spec doesn't need change.

### CODE REVIEW

File: packages/files-ui/src/Components/Modules/Settings/Profile.tsx [hunk 1]
```diff
@@ -242,42 +242,48 @@ const ProfileView = () => {
...
-          error: ""
-        })
-      }
-    }).catch(console.error)
+    lookupOnUsername(username)
+      .then((doesUsernameExist) => {
+        if (doesUsernameExist) {
+          setUsernameData({
+            loading: false,
+            error: "This username is already taken"
```
Comment (tanmoyAtb) [inline packages/files-ui/src/Components/Modules/Settings/Profile.tsx]: missed trans here!
```suggestion
            error: t`This username is already taken`
```

File: packages/files-ui/src/Contexts/UserContext.tsx [hunk 2]
```diff
@@ -116,16 +116,16 @@ const UserProvider = ({ children }: UserContextProps) => {
...
       )
     }
   }
 
   const lookupOnUsername = async (username: string) => {
     if (!profile) return false
     try {
-      await filesApiClient.lookupUser({ username })
-      return true
+      const alreasdyExists = await filesApiClient.lookupUser({ username })
```
Comment (tanmoyAtb) [inline packages/files-ui/src/Contexts/UserContext.tsx]:
```suggestion
      const alreadyExists = await filesApiClient.lookupUser({ username })
```

File: packages/files-ui/src/Contexts/UserContext.tsx [hunk 3]
```diff
@@ -116,16 +116,16 @@ const UserProvider = ({ children }: UserContextProps) => {
...
+      return !!alreasdyExists
```
Comment (tanmoyAtb) [inline packages/files-ui/src/Contexts/UserContext.tsx]:
```suggestion
      return !!alreadyExists
```

File: packages/files-ui/src/Components/Modules/Settings/Profile.tsx [hunk 4]
```diff
@@ -242,42 +242,48 @@ const ProfileView = () => {
...
   const debouncedOnLookupUsername = useMemo(
     () => debounce(onLookupUsername, 300),
     [onLookupUsername]
   )
 
-  const onUsernameChange = (value: string) => {
-    setUsername(value)
-    debouncedOnLookupUsername(value)
+  const onUsernameChange = (value: string | number | undefined) => {
+    const sanitizedValue = value?.toString() || ""
```
Comment (tanmoyAtb) [inline packages/files-ui/src/Components/Modules/Settings/Profile.tsx:283]: This makes me wonder, if we made the same mistake in other inputs!

**Review by tanmoyAtb — APPROVED** [review]
Make sure to check the suggestions!

File: packages/files-ui/src/Components/Modules/Settings/Profile.tsx (hunk 4 above)
Comment (Tbaut) [reply packages/files-ui/src/Components/Modules/Settings/Profile.tsx:283]: Not sure, we can check.

File: packages/files-ui/src/locales/en/messages.po [hunk 5]
```diff
@@ -448,8 +451,8 @@ msgstr "Remind me later"
 msgid "Rename"
 msgstr "Rename"
 
-msgid "Rename File/Folder"
-msgstr "Rename File/Folder"
+msgid "Rename ${0}"
```
Comment (Tbaut) [inline packages/files-ui/src/locales/en/messages.po]: pretty sure this won't fly, I'll check it out.

File: packages/files-ui/src/Components/Modules/Settings/Profile.tsx (hunk 4 above)
Comment (asnaith) [reply packages/files-ui/src/Components/Modules/Settings/Profile.tsx:283]: I checked these inputs in the UI:

- change password inputs
- first and last name inputs
- file rename input (list and grid view)
- folder rename input (list and grid view)
- create folder input

I didn't have a problem removing anything that was entered, none of them exhibited the same issue that was happening on the username input.

**Review by RyRy79261 — APPROVED** [review]

**Pull request merged**